### PR TITLE
fix(metrics-api): Only paginate with orderBy

### PR DESCRIPTION
**Pull request by jjbayer**
We disallow pagination parameters `per_page` and `cursor` except when there is also an `orderBy`.
But the default `per_page` of 50 was still accidentally applied as a limit in any other case, leading to incomplete time series filled with zeros, e.g.

```
460,
0,
0,
449,
0,
0,
405,
412,
```

instead of 

```
460,
479,
458,
449,
421,
394,
405,
412,
```

With this PR, the default limit provided by the paginator is now ignored unless there is an `orderBy`.

### CODE REVIEW

File: src/sentry/snuba/metrics/helpers.py [hunk 1]
```diff
@@ -220,35 +220,28 @@ def _parse_orderby(self, query_params):
...
                 # possible
                 raise InvalidParams("'per_page' is only supported in combination with 'orderBy'")
-
-        if limit is not None:
-            try:
-                limit = int(limit)
-                if limit < 1:
-                    raise ValueError
-            except (ValueError, TypeError):
-                raise InvalidParams("'limit' must be integer >= 1")
```
Comment (jjbayer) [inline src/sentry/snuba/metrics/helpers.py:238]: Unrelated cleanup: `limit` is already validated by the paginator so no need to do it here.

**Review by matejminar — APPROVED** [review]

**Review by ahmedetefy — APPROVED** [review]
